### PR TITLE
bpo-32642: adding compatibility for pathlike objects in sys.path

### PR DESCRIPTION
**Pull request by jayyyin**
I'm uncertain if that's all that needs to be done for the code to do what is intended since I'm kind of new to the whole path-like object concept, I've noticed that there's still self.path in the class and recall reading something about removing path attributes from classes, but I've been unable to find documentation on how to add compatibility to something for path-like objects. 

https://bugs.python.org/issue32642

### CODE REVIEW

**Comment by jayyyin** [comment]
I'm having issues with my local changes for this PR, I'm unsure why my local machine takes a seemingly infinite amount of time on test_poplib, so again I think something to do with my local environment is causing issues again, any help would be appreciated...

**Comment by jayyyin** [comment]
https://pastebin.com/q4FKnPZH

the trace for the test_poplib

**Comment by csabella** [comment]
@jayyyin Please resolve the file conflict.

File: Lib/importlib/_bootstrap_external.py [hunk 1]
```diff
@@ -1241,6 +1241,8 @@ def _get_spec(cls, fullname, path, target=None):
         #  the list of paths that will become its __path__
         namespace_path = []
         for entry in path:
+            if hasattr(entry, '__fspath__'):
```
Comment (brettcannon) [inline Lib/importlib/_bootstrap_external.py:1244]: This is unnecessary as `os.fspath()` passes strings through without issue.

File: Misc/NEWS.d/next/Library/2018-02-15-12-52-34.bpo-32642.e4g57h.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+adding compatibility for pathlike objects in sys.path
```
Comment (brettcannon) [inline Misc/NEWS.d/next/Library/2018-02-15-12-52-34.bpo-32642.e4g57h.rst:1]:
```suggestion
Add compatibility for path-like objects on sys.path.
Patch by jayyin.
```

**Review by brettcannon — CHANGES_REQUESTED** [review]
Thanks for the PR but this needs tests.

**Comment by csabella** [comment]
The original author of the pull request hasn't been active on this issue for a long time, so I'm going to close it.  This can be reopened or a new pull request can be created for work to continue on the issue.  Thanks!

**Pull request closed without merging**